modules/reactor/src/meshgenerators/CoarseMeshExtraElementIDGenerator.C

```diff
@@ -0,0 +1,103 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CoarseMeshExtraElementIDGenerator.h"
+
+registerMooseObject("ReactorApp", CoarseMeshExtraElementIDGenerator);
+
+#include "libmesh/enum_point_locator_type.h"
+#include "libmesh/elem.h"
+#include "libmesh/mesh_serializer.h"
+
+InputParameters
+CoarseMeshExtraElementIDGenerator::validParams()
+{
+  InputParameters params = MeshGenerator::validParams();
+  params.addRequiredParam<MeshGeneratorName>(
+      "input", "Name of an existing mesh generator to which we assign coarse element IDs");
+  params.addRequiredParam<MeshGeneratorName>(
+      "coarse_mesh", "Name of an existing mesh generator as the coarse mesh");
+  params.addRequiredParam<std::string>("coarse_element_id_name", "Name for the coarse element ID");
+  params.addParam<bool>("enforce_mesh_embedding",
+                        false,
+                        "True to error out when the input mesh is not embedded in the coarse mesh");
+
+  params.addClassDescription("Assign coarse element IDs for elements on a "
+                             "mesh based on a coarse mesh.");
+  return params;
+}
+
+CoarseMeshExtraElementIDGenerator::CoarseMeshExtraElementIDGenerator(const InputParameters & params)
+  : MeshGenerator(params),
+    _input(getMesh("input")),
+    _coarse_mesh(getMesh("coarse_mesh")),
+    _coarse_id_name(getParam<std::string>("coarse_element_id_name")),
+    _must_embedding(getParam<bool>("enforce_mesh_embedding"))
+{
+}
+
+std::unique_ptr<MeshBase>
+CoarseMeshExtraElementIDGenerator::generate()
+{
+  std::unique_ptr<MeshBase> mesh = std::move(_input);
+
+  std::unique_ptr<MeshBase> coarse_mesh = std::move(_coarse_mesh);
```

Comment:
Not sure. Anyone has an answer? If not, I will keep this.

Reply:
I called Logan. It's done to semantically move the ownership to this routine of the unique pointer. Like we dont want to be acting on _input directly because it'd look weird.
Since there is an implicit copy at return the ownership is nicely moved from mesh generator to mesh generator already anyway

So no need (and let's not) delete it.

Reply:
Sounds good. I need to spend some time to educate myself on this `move` and smart pointer stuff.